### PR TITLE
Add lazy find, finds, exists  #58 & 54

### PR DESCRIPTION
**Pull request by TroyWalshProf**
Add lazy find and finds #58
Add exists to lazy element #54

### CODE REVIEW

File: Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+        /// Gets a value indicating whether the lazy element is selected
+        /// </summary>
+        public bool Selected
+        {
+            get
+            {
+                this.Log.LogMessage(MessageType.INFORMATION, $"Checking to see if the lazy element '{this.userFriendlyName}' is selected");
+
+                return GenericWait.Wait<bool>(
+                    () =>
```
Comment (FermJacob) [inline Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs]: Can we fix the formatting?

File: Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+
+        /// <summary>
+        /// Wait for and get the visible web element
+        /// </summary>
+        /// <returns>The web visible web element</returns>
+        public IWebElement GetTheVisibleElement()
+        {
+            if (this.elementIndex == null)
+            {
+                this.CachedElement = (this.parent == null) ? this.WebDriver.Wait().ForVisibleElement(this.By) :
```
Comment (FermJacob) [inline Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs]: Do we want to perform any logging here?

File: Framework/NugetBuild/package.nuspec [hunk 3]
```diff
@@ -13,6 +13,12 @@
     <iconUrl>http://magenicautomation.azurewebsites.net/content/favicon.ico</iconUrl>
     <summary>Magenic's automation quick start framework. MAQS is a modular test automation framework that can be used as the base for your automation project or individual pieces can be used to enhance existing frameworks. Utilizing project templates, NuGet, and utility libraries we are able to have a functioning framework up and running in minutes, almost entirely removing on the initial time investment on implementing an automation solution.</summary>
     <releaseNotes>
+      5.7.0 - 2019/11/??????
```
Comment (FermJacob) [inline Framework/NugetBuild/package.nuspec]: I'm going to let this one go 👍

File: Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs (hunk 1 above)
Comment (TroyWalshProf) [reply Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs]: Not seeing the problem.  Can you be more specific about what you expect/want to see?

Comment (FermJacob) [reply Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs]: The start of the function () is misaligned for a few methods in this class

```
return GenericWait.Wait<bool>(
                    () =>
                    {
```

``` 
return GenericWait.Wait<bool>(
                    () =>
                {
```

Comment (TroyWalshProf) [reply Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs]: This the way the IDE natively formats this kind of call, so standardizing on this format

File: Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs (hunk 2 above)
Comment (TroyWalshProf) [reply Framework/BaseSeleniumTest/Extensions/AbstractLazyIWebElement.cs]: Nope, this logging comes with "GetElement".  Adding additional logging here would make everything (the majority of things) create dup log entries.

File: Framework/NugetBuild/package.nuspec (hunk 3 above)
Comment (TroyWalshProf) [reply Framework/NugetBuild/package.nuspec]: I can change this if you want to pick a date for this release, but I don't know when we will  be releasing

**Pull request merged**